exp/api/v1beta1/gcpmanagedcontrolplane_types.go

```diff
@@ -142,6 +146,25 @@ type MasterAuthorizedNetworksConfigCidrBlock struct {
 	CidrBlock string `json:"cidr_block,omitempty"`
 }
 
+// NetworkPolicy represents configuration options for NetworkPolicy feature of the GKE cluster.
+type NetworkPolicy struct {
+	// The selected network policy provider.
+	// +optional
+	Provider *NetworkPolicyProvider `json:"provider,omitempty"`
+	// Whether network policy is enabled on the cluster.
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
```

Comment:
I don't think we need **Enabled** in the API. If someone wants to enable NetworkPolicy i think having a non-nil **NetworkPolicy** is sufficient.